.github/workflows/build-all.yml

```diff
@@ -0,0 +1,241 @@
+          echo "::set-output name=n1_synced::$SYNC_STATE_N1"
+          echo "::set-output name=n2_synced::$SYNC_STATE_N2"
+          echo "::set-output name=n3_synced::$SYNC_STATE_N3"
+          echo "::set-output name=n4_synced::$SYNC_STATE_N4"
+          echo "::set-output name=n1_freshness::$FRESHNESS_N1"
+          echo "::set-output name=n2_freshness::$FRESHNESS_N2"
+          echo "::set-output name=n3_freshness::$FRESHNESS_N3"
+          echo "::set-output name=n4_freshness::$FRESHNESS_N4"
+          echo "::set-output name=n1_unreachable::$UNREACHABLE_N1"
+          echo "::set-output name=n2_unreachable::$UNREACHABLE_N2"
+          echo "::set-output name=n3_unreachable::$UNREACHABLE_N3"
+          echo "::set-output name=n4_unreachable::$UNREACHABLE_N4"
+
+      - name: Safe ledger_state.json for later inspection
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.os_version }}-tmp-test-${{ github.sha }}-ledger_state.json
+          path: ledger_state.json
+          retention-days: 3
+
+      - name: Fail if ledger is not in sync
+        if: |
+          steps.ledger.outputs.n1_synced != 'true' ||
+          steps.ledger.outputs.n2_synced != 'true' ||
+          steps.ledger.outputs.n3_synced != 'true' ||
+          steps.ledger.outputs.n4_synced != 'true' ||
+          steps.ledger.outputs.n1_freshness != 'true' ||
+          steps.ledger.outputs.n2_freshness != 'true' ||
+          steps.ledger.outputs.n3_freshness != 'true' ||
+          steps.ledger.outputs.n4_freshness != 'true' ||
+          steps.ledger.outputs.n1_unreachable != 0 ||
+          steps.ledger.outputs.n2_unreachable != 0 ||
+          steps.ledger.outputs.n3_unreachable != 0 ||
+          steps.ledger.outputs.n4_unreachable != 0
+
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('${{ matrix.os_version }} - Not all nodes are in sync!')
+
+
+  push_to_registry:
+    name: Push Docker image to GitHub Packages
+    runs-on: ubuntu-latest
+    needs: build_and_test
+    strategy:
+      matrix:
+        os_version: [
+          'buster',
+          'ubuntu16',
+          'ubuntu18'
+        ]
+
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.os_version }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ matrix.os_version }}
+            ${{ runner.os }}-buildx
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Lowercase repo owner
```

Comment:
`Improvement`: Could be replaced by this GHA
https://github.com/ASzc/change-string-case-action

Reply:
I wouln't do that that would feeling for me like the lodash approach in CI CD Pipelines.

Reply:
I will leave this for now, the bash command works fine.